honeybee_vtk/scene.py

```diff
@@ -1,281 +1,177 @@
-
-    @staticmethod
-    def _check_tuple(bg_color):
-        """Check if all values in the tuple are integers.
-
-        Args:
-            bg_color: User input for background color
-
-        Returns:
-            A boolean value if True or None.
-        """
-        item_check = [isinstance(v, int) for v in bg_color]
-        if item_check.count(True) == 3:
-            return True
-
-    def _create_render_window(self, background_color=None) \
-            -> Tuple[
-                vtk.vtkRenderWindowInteractor, vtk.vtkRenderWindow, vtk.vtkRenderer
-            ]:
-        """Create a rendering window with a single renderer and an interactor.
-
-        The objects are embedded inside each other:
-        interactor is the outmost layer. A render is added to a window and then the
-        window is set inside the interactor.
-
-        If you are thinking why, the answer is that is how VTK is designed to work.
-
-        Args:
-            background_color: A tuple of three floats that represent RGB values of the
-                color that you'd like to set as the background color. Defaults to None.
-
-        Returns:
-            Tuple -- window_interactor, render_window, renderer
-        """
-        renderer = vtk.vtkRenderer()
-        window = vtk.vtkRenderWindow()
-        interactor = vtk.vtkRenderWindowInteractor()
-        # add renderer to rendering window
-        window.AddRenderer(renderer)
-        # set rendering window in window interactor
-        interactor.SetRenderWindow(window)
-
-        # Validate background color and set it for the render window
-        if not background_color:
+    def __init__(self, background_color: Tuple[int, int, int] = None) -> None:
+        self.background_color = background_color
+        self._actors = {}
+        self._cameras = []
+        self._windows = []
+
+    @property
+    def background_color(self) -> Tuple[int, int, int]:
+        """background_color for the scene."""
+        return self._background_color
+
+    @background_color.setter
+    def background_color(self, val: Tuple[int, int, int]) -> None:
+        if not val:
             colors = vtk.vtkNamedColors()
-            background_color = colors.GetColor3d("SlateGray")
-
-        elif isinstance(background_color, tuple) and len(background_color) == 3\
-                and self._check_tuple(background_color):
-            pass
+            self._background_color = colors.GetColor3d("SlateGray")
+        elif _validate_input(val, int):
+            self._background_color = val
         else:
             raise ValueError(
                 'Background color is a tuple with three integers'
                 ' representing R, G, and B values.'
             )
 
-        renderer.SetBackground(background_color)
-        renderer.TwoSidedLightingOn()
+    @property
+    def actors(self) -> Dict[str, Actor]:
+        """A dictionary of vtk actor name: vtk actor structure."""
+        return self._actors
```

Comment:
Why do we need a dictionary? Can't we just have the name as a property of the Actor itself? That will be more elegant in my opinion.

Reply:
Complied.